src/Scoper.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the webmozart/php-scoper package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Webmozart\PhpScoper;
+
+use PhpParser\Node\Name;
+use PhpParser\Node\Stmt\Namespace_;
+use PhpParser\NodeTraverser;
+use PhpParser\Parser;
+use PhpParser\PrettyPrinter\Standard;
+use Webmozart\PhpScoper\NodeVisitor\FullyQualifiedNamespaceUseScoperNodeVisitor;
+use Webmozart\PhpScoper\NodeVisitor\NamespaceScoperNodeVisitor;
+use Webmozart\PhpScoper\NodeVisitor\UseNamespaceScoperNodeVisitor;
+
+class Scoper
+{
+    /**
+     * @var Parser
+     */
+    private $parser;
+
+    public function __construct(Parser $parser)
+    {
+        $this->parser = $parser;
+    }
+
+    /**
+     * @param $content
+     * @param $prefix
+     *
+     * @return string
+     */
+    public function scope($content, $prefix)
+    {
+        $traverser = new NodeTraverser();
+        $traverser->addVisitor(new NamespaceScoperNodeVisitor($prefix));
+        $traverser->addVisitor(new UseNamespaceScoperNodeVisitor($prefix));
+        $traverser->addVisitor(new FullyQualifiedNamespaceUseScoperNodeVisitor($prefix));
+
+        //TODO Manage errors
+        $statements = $this->parser->parse($content);
+        $statements = $traverser->traverse($statements);
+
+        $prettyPrinter = new Standard();
+
+        return $prettyPrinter->prettyPrintFile($statements)."\n";
```

Comment:
I think that instead of changing `$content`, you should collect all the offsets where you want to insert the namespace prefix and them insert them into the original content in one go. That way you can maintain the original file format.

See also https://github.com/webmozart/php-scoper/issues/1#issuecomment-148896044.